### ISSUE TITLE
How to train my own dataset

### ISSUE THREAD

**Issue by wei-yuan** [matterport/Mask_RCNN#408]
Hello everyone

First, it's a very detailed repo, thanks for everything the contributors have done here.

These days, I am working on training with my own datasets, but I get stuck on few questions.  

I have read "Training on Your Own Dataset" section,  `train_shapes.ipynb` and `inspect_data.ipynb`.

Here's what I have now

### Number of class
Only one
### Images
It contains only one class of images, for example, a ball like stuff
### Masks
I already have its mask images as ground truth.
---
## Questions
In "Training on Your Own Dataset" section 
### Q1
About Config, it said 
> Subclass it and modify the attributes you need to change. 

Does this mean that I should create a subclass of class `Config`(which is in `config.py`) in a python file like `coco.py`? 

### Q2
In the training part of [train_shapes.ipynb](https://github.com/matterport/Mask_RCNN/blob/master/train_shapes.ipynb),  what is inside `dataset_train` and `dataset_val` object in the snippet below? I'm not sure whether should I feed images with `bbox` intio the model if I use my own `Dataset` class?

    model.train(dataset_train, dataset_val, 
                learning_rate=config.LEARNING_RATE, 
                epochs=1, 
                layers='heads')
---
If there is any training process document that I missed, pls tell me

I would be thankful for any suggestion, thank you for your time :)

**Comment by waleedka** [comment]
Yes, you need to create a subclass of `Config`. And, your dataset needs to feed images and masks to the model. 

See this post, which goes into it in more detail:
https://engineering.matterport.com/splash-of-color-instance-segmentation-with-mask-r-cnn-and-tensorflow-7c761e238b46

And take a look at the examples in the `samples` directory. (get latest update from github first as we just reorganized the directory structure).

**Comment by wei-yuan** [comment]
＠waleedka It's helpful. Thank you for your time!

**Comment by guohualee** [comment]
@wei-yuan I also face this problem. ALL my dataset have are original images and the corespongding mask images, and have only one class. I am confused what to feed to the model,can you pls give some advice?
